src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5067,44 +5067,26 @@ public override IOperation VisitAnonymousObjectCreation(IAnonymousObjectCreation
             ImplicitInstanceInfo savedCurrentImplicitInstance = _currentImplicitInstance;
             _currentImplicitInstance = new ImplicitInstanceInfo((INamedTypeSymbol)operation.Type);
 
-            var properties = operation.Type.GetMembers().OfType<IPropertySymbol>().ToImmutableArray();
-            Debug.Assert(properties.Length == operation.Initializers.Length);
-
             SpillEvalStack();
 
             var initializerBuilder = ArrayBuilder<IOperation>.GetInstance(operation.Initializers.Length);
             for (int i = 0; i < operation.Initializers.Length; i++)
             {
-                IOperation initializer = operation.Initializers[i];
-                IPropertySymbol initializedProperty = properties[i];
+                var simpleAssignment = (ISimpleAssignmentOperation)operation.Initializers[i];
```

Comment:
>(ISimpleAssignmentOperation)operation.Initializers[i]; [](start = 39, length = 54)

Since we cannot change type of the ```initializers``` property. I think we should clarify the contract in its doc comment and assert the invariant in two our test visitors.